File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/GameEndedEvent.java
package dev.robocode.tankroyale.botapi.events;

import dev.robocode.tankroyale.botapi.BotResults;

/**
 * Event occurring when game has just ended.
 */
@SuppressWarnings("unused")
public final class GameEndedEvent implements IEvent {

    // Number of rounds played.
    private final int numberOfRounds;

    // Results of the battle.
    private final BotResults results;

    /**
     * Initializes a new instance of the GameEndedEvent class.
     *
     * @param numberOfRounds is the number of rounds played.
     * @param results        is the bot results of the battle.
     */
    public GameEndedEvent(int numberOfRounds, BotResults results) {
        this.numberOfRounds = numberOfRounds;
        this.results = results;
    }

    /**
     * Returns the number of rounds played.
     *
     * @return The number of rounds played.
     */
    public int getNumberOfRounds() {
        return numberOfRounds;
    }

    /**
     * Returns the results of the battle.
     *
     * @return The results of the battle.
     */
    public BotResults getResults() {
        return results;
    }
}

File: bot_api_py/events/game_ended_event.py
from bot_api_py.bot_results import BotResults
from bot_api_py.events.abstract_event import IEvent


class GameEndedEvent(IEvent):
    """Event occurring when game has just ended."""

    def __init__(self, number_of_rounds: int, results: BotResults):
        """Initializes a new instance of the GameEndedEvent class.

        Args:
            number_of_rounds: The number of rounds played.
            results: The bot results of the battle.
        """
        self.number_of_rounds = number_of_rounds
        self.results = results

    def get_number_of_rounds(self) -> int:
        """Returns the number of rounds played.

        Returns:
            The number of rounds played.
        """
        return self.number_of_rounds

    def get_results(self) -> BotResults:
        """Returns the results of the battle.

        Returns:
            The results of the battle.
        """
        return self.results